### PR TITLE
[4.0] Add checkall-toggle to template styles

### PR DESCRIPTION
**Pull request by hardik-codes**
This PR adds checkall-toggle to template styles so that it is consistent with other components

### Testing Instructions
Open template styles

### Expected result
checkall-toggle_new

### Actual result
checkall-toggle_old

### Documentation Changes Required
None

### CODE REVIEW

File: administrator/components/com_templates/tmpl/styles/default.php [hunk 1]
```diff
@@ -37,7 +37,7 @@
 						<thead>
 							<tr>
 								<td style="width:1%" class="text-center">
-									&#160;
+									<input type="checkbox" name="checkall-toggle" value="" title="<?php echo Text::_('JGLOBAL_CHECK_ALL'); ?>" onclick="Joomla.checkAll(this)">
```
Comment (Quy) [inline administrator/components/com_templates/tmpl/styles/default.php]:
```suggestion
									<?php echo HTMLHelper::_('grid.checkall'); ?>
```

**Comment by hardik-codes** [comment]
done @Quy

**Comment by hardik-codes** [comment]
@alikon @infograf768 @Quy please test this PR

**Comment by infograf768** [comment]
This PR does not take into account the Toolbar
"Default"

If checkall is used, it should not be available i.e should be greyed

**Comment by hardik-codes** [comment]
@infograf768 
But in other components also like, com_menu, it is not "greyed" 
toggle-checkall

Isn't this an expected behavior?

**Comment by infograf768** [comment]
The problem Here is that `listCheck` is boolean, i.e. we can't apparently force selecting only one item.
And, yes, I think we have a similar problem for com_menus when the menu items selected are assigned the same language (or ALL).

**Comment by infograf768** [comment]
For menus, we have an error
Screen Shot 2019-04-13 at 11 52 02

**Comment by hardik-codes** [comment]
@infograf768 what could be a possible solution then?

**Comment by infograf768** [comment]
I see that if one choses multiple template styles and use the Default button, then it is the one with the smaller id which is set as default. No message.

**Comment by infograf768** [comment]
Therefore, I would just forget adding a checkall for template styles. It is too imprevisible.
We also don't do it for Languages.

**Comment by hardik-codes** [comment]
Ok, I get your point, then How do I grey out Toolbar "Default", if check all is used?

**Comment by infograf768** [comment]
As I said, I don’t think it is necessary. We don’t really need to use a checkall in this manager

**Comment by hardik-codes** [comment]
But for the design of this component to be consistent with others, shouldn't this checkall be included?

**Comment by brianteeman** [comment]
The component should have the checkall.I have seen sites with several pages of template styles. And of course for a consistent ui.

@infograf768 is also correct that the "default" button should be disabled when multiple items are checked.

**Comment by hardik-codes** [comment]
@brianteeman @infograf768 @bembelimen is there any component in which a toolbar button is disabled when multiple items are selected(to be used as a reference)

**Comment by brianteeman** [comment]
the toolbar has changed a lot in j4 so I'm not sure - you would have to check. I think in j3 the  "home" button in the menu manager has the "check" functionality you are looking for although it didnt disable the button and showed a message instead

**Comment by infograf768** [comment]
> is there any component in which a toolbar button is disabled when multiple items are selected(to be used as a reference)

no. see above a screen shot of what we get for menu items.
I have also explained above the boolean aspect of “listCheck” which allows greying the button when nothing is selected but does not consider the number of ticked items.

One could create a specific query in the model throwing an alert when attempting to create multiple default styles.

**Comment by infograf768** [comment]
We can at least get an alert message by using
```
			$toolbar->confirmButton('default')
				->text('COM_TEMPLATES_TOOLBAR_SET_HOME')
				->message('COM_TEMPLATES_CONFIRM_SET_DEFAULT')
				->task('styles.setDefault')
				->listCheck(true);
```
The message would contain a text telling the user to make sure a unique style is chosen to be set as default.

Screen Shot 2019-04-14 at 07 41 53

**Comment by hardik-codes** [comment]
> We can at least get an alert message by using

@infograf768 which class should I include, to use ```confirmButton```?

**Comment by infograf768** [comment]
use Joomla\CMS\Toolbar\Toolbar;

an then in the method

// Get the toolbar object instance
		$toolbar = Toolbar::getInstance('toolbar');

**Comment by hardik-codes** [comment]
@infograf768 having done that the "default" button is not disabled even if no item is checked?

**Comment by infograf768** [comment]
btw, this could be done in another PR as using the Toolbar class only for that Toolbar button would be a waist. Also, I have not figured yet how to add an icon there.

**Comment by infograf768** [comment]
Please let's separate the stuff with the view...

File: administrator/components/com_templates/View/Styles/HtmlView.php [hunk 2]
```diff
@@ -123,7 +127,13 @@ protected function addToolbar()
 
 		if ($canDo->get('core.edit.state'))
 		{
-			ToolbarHelper::makeDefault('styles.setDefault', 'COM_TEMPLATES_TOOLBAR_SET_HOME');
+			$toolbar->confirmButton('setDefault')
```
Comment (infograf768) [inline administrator/components/com_templates/View/Styles/HtmlView.php]: To get the icon, it should be
$toolbar->confirmButton('default')

**Comment by hardik-codes** [comment]
@infograf768 I was able to add the icon 
Also, I have included the changes suggested by you

**Comment by infograf768** [comment]
As I said, the view needs more changes once we use the Toolbar class.
Let's make this in a separate PR

**Comment by hardik-codes** [comment]
> As I said, the view needs more changes once we use the Toolbar class.
> Let's make this in a separate PR

Could you elaborate for what, the separate PR should be?

**Comment by infograf768** [comment]
> Could you elaborate for what the separate PR should be?

It would be ONLY for the View.
It would modify most of `ToolbarHelper::` to use `$toolbar->`
I am on it now. Trying to find the way to also do it for duplicate.
In any case we do need a new string also for the Default message.

**Comment by hardik-codes** [comment]
> > Could you elaborate for what the separate PR should be?
> 
> It would be ONLY for the View.
> It would modify most of `ToolbarHelper::` to use `$toolbar->`
> I am on it now. Trying to find the way to also do it for duplicate.
> In any case we do need a new string also for the Default message.

Should I revert the latest commit?

**Comment by infograf768** [comment]
> Should I revert the latest commit?

Yep. Let's keep this PR for its original purpose.

**Comment by hardik-codes** [comment]
@infograf768 kindly mark your test for this PR

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on fc20ee4bcbe9cbc0f4be7c18ba64d8ca6c58b812<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24541">issues.joomla.org/tracker/joomla-cms/24541</a>.</sub>

**Comment by infograf768** [comment]
Will make the pr tomorrow for the view.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 429d0c7c05bfebbce2ca442af2fb06ab66f006b1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24541">issues.joomla.org/tracker/joomla-cms/24541</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24541">issues.joomla.org/tracker/joomla-cms/24541</a>.</sub>

**Comment by infograf768** [comment]
Please test https://github.com/joomla/joomla-cms/pull/24591 concerning the alert.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**